lib/manageiq/providers/hawkular/engine.rb

```diff
@@ -3,6 +3,11 @@ module Providers
     module Hawkular
       class Engine < ::Rails::Engine
         isolate_namespace ManageIQ::Providers::Hawkular
+
+        initializer('hawkular.autoload') do |app|
+          gem_root = File.expand_path('../../../..', __dir__)
+          config.autoload_paths += Dir["#{gem_root}/{app,lib}/**/"]
```

Comment:
@durandom can this be put in the provider generator?  Do we want to autoload all directories under lib?  

Isn't app already autoloaded @cfcosta  for engines?

Reply:
Nope, not that I'm aware of. I'll do some more testing, because it seems that only UI changes are actually autoloaded, if any. Would be good to have that enabled for workers as well.

Reply:
`ActiveSupport::Dependencies.autoload_paths` includes the `app/models` directory of every provider.

If you need `lib` or anything else, I suggest to add it like the UI does [here](https://github.com/ManageIQ/manageiq-ui-classic/blob/5fac2d1037453493187a6df0c4104e8b4fc0047f/lib/manageiq/ui/classic/engine.rb#L34)